### PR TITLE
6007 and 6017

### PR DESCRIPTION
**Pull request by Sadibhatla**
## 🎫 Ticket

https://jira.cms.gov/browse/AB2D-6007
https://jira.cms.gov/browse/AB2D-6017

## 🛠 Changes

 Changed default values from boolean to Null
 Updated opt_out_flag to look for everything else than "False" i.e. true or null
 Updated 2024 proc to remove sandbox partition 
 Removed 2023 proc
 Added new columns to current_mbi table
 Created a new simple view to have mbi as a column and retriving partitions from 2020-23 data from there

## ℹ️ Context for reviewers

(Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.)

## ✅ Acceptance Validation

(How were the changes verified? Did you fully test the acceptance criteria in the ticket? Provide reproducible testing instructions and screenshots if applicable.)

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

**Review by smirnovaae — APPROVED** [review]

File: common/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -167,3 +167,5 @@ databaseChangeLog:
    #     file: db/changelog/v2023/create_pg_cron_extension.sql
   - include:
       file: db/changelog/v2024/create_current_mbi_procedure.sql
+  - include:
```
[the remark this answers is not in the document]
Comment (Sadibhatla) [reply common/src/main/resources/db/changelog/db.changelog-master.yaml:170]: image

[the remark this answers is not in the document]
Comment (Sadibhatla) [reply common/src/main/resources/db/changelog/db.changelog-master.yaml:170]: Drop procs are there as part of  misc sql file itself at the end

File: common/src/main/resources/db/changelog/v2024/misc_opt_out_attribution.sql [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+AS $$
+begin
+update current_mbi  set mbi =sub.current_mbi,effective_date=sub.effective_date,opt_out_flag=sub.opt_out_flag
+    from
+(SELECT distinct current_mbi,effective_date,opt_out_flag
+FROM coverage_view_with_mbi
+WHERE contract  in (
+SELECT DISTINCT(contract_number)
+                       FROM job_view
+                      WHERE contract_number not LIKE 'Z%' --OR contract_number LIKE 'E%'
```
[the remark this answers is not in the document]
Comment (Sadibhatla) [reply common/src/main/resources/db/changelog/v2024/misc_opt_out_attribution.sql:46]: Yes.. I agree !

File: common/src/main/resources/db/changelog/db.changelog-master.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (smirnovaae) [reply common/src/main/resources/db/changelog/db.changelog-master.yaml:170]: > Drop procs are there as part of misc sql file itself at the end

ah! missed it. wonderful!

File: common/src/main/resources/db/changelog/v2024/misc_opt_out_attribution.sql [hunk 3]
```diff
@@ -11,32 +11,39 @@ SELECT cov.contract,
...
+insert into public.current_mbi(
+    mbi, effective_date, opt_out_flag)
+SELECT distinct current_mbi,effective_date,opt_out_flag
+FROM coverage_view_with_mbi
+WHERE contract  in (
+    SELECT DISTINCT(contract_number)
+    FROM job_view
+    WHERE contract_number not LIKE 'Z%'
+) and effective_date> CURRENT_DATE - 1
+on conflict do nothing
```
Comment (smirnovaae) [inline common/src/main/resources/db/changelog/v2024/misc_opt_out_attribution.sql]: please add `;` in the end 
and then 
`end;
$$`

**Pull request merged**